Source/Rio.API/Controllers/ParcelLedgerController.cs

```diff
@@ -33,11 +34,22 @@ public IActionResult New([FromBody] ParcelLedgerCreateDto parcelLedgerCreateDto)
             var parcelDto = Parcel.GetByParcelNumberAsDto(_dbContext, parcelLedgerCreateDto.ParcelNumbers[0]); 
             if (parcelDto == null)
             {
-                ModelState.AddModelError("ParcelNumber", $"{parcelLedgerCreateDto.ParcelNumbers[0]} is not a valid Parcel APN.");
-                return BadRequest(ModelState);
+                if (parcelLedgerCreateDto.ParcelNumbers[0] == null)
+                {
+                    ModelState.AddModelError("ParcelNumber", "The Parcel APN field is required.");
+                }
+                else
+                {
+                    ModelState.AddModelError("ParcelNumber", $"{parcelLedgerCreateDto.ParcelNumbers[0]} is not a valid Parcel APN.");
+                }
+            }
+
+            if (parcelLedgerCreateDto.TransactionTypeID == (int) TransactionTypeEnum.Supply && parcelLedgerCreateDto.WaterTypeID == null)
+            {
+                ModelState.AddModelError("SupplyType", "The Supply Type field is required for transactions adjusting water supply.");
             }
 
-            ValidateEffectiveDate(parcelLedgerCreateDto.EffectiveDate);
+            ValidateEffectiveDate(parcelLedgerCreateDto.EffectiveDate.Value, parcelLedgerCreateDto.EffectiveDateString);

```

Comment:
why is there a string and a date version?  why are we using both?

Reply:
I need to validate the user-inputted string to make sure it's in the right YYYY-MM-DD format, but I can just pass the string and then parse it as a date after it's validated.

Reply:
EffectiveDateString field removed, EffectiveDate is now a string and format is validated via RegularExpressionAttribute before parsing as DateTime.